core/src/main/java/io/parsingdata/metal/token/Nod.java

```diff
@@ -38,9 +38,20 @@ public Nod(final ValueExpression size, final Encoding enc) {
     @Override
     protected ParseResult parseImpl(final String scope, final Environment env, final Encoding enc) throws IOException {
         final OptionalValueList ov = _size.eval(env, enc);
-        if (ov.isEmpty()) { return new ParseResult(false, env); }
-        if (ov.size > 1) { throw new IllegalStateException("Size may not evaluate to more than a single value."); }
-        return ov.head.isPresent() ? new ParseResult(true, new Environment(env.order, env.input, env.offset + ov.head.get().asNumeric().longValue())) : new ParseResult(false, env);
+        if (ov.isEmpty()) {
```

Comment:
@jvdb Is this `ov.isEmpty()` check actually needed now, by the way? I see that in `Def` the check is slightly different:

```java
        final OptionalValueList size = _size.eval(env, enc);
        if (size.size != 1) { throw new IllegalStateException("Size may not evaluate to more than a single value."); }
        if (!size.head.isPresent()) {
            return new ParseResult(false, env);
        }
        // TODO: Handle value expression results as BigInteger (#16)
        final int dataSize = size.head.get().asNumeric().intValue();
        if (dataSize < 0) {
            return new ParseResult(false, env);
        }
```